pkg/hub/manager.go

```diff
@@ -82,11 +83,20 @@ func RunControllerManager(ctx context.Context, controllerContext *controllercmd.
 		controllerContext.EventRecorder,
 	)
 
-	csrController := csr.NewCSRApprovingController(
-		kubeClient,
-		kubeInfomers.Certificates().V1().CertificateSigningRequests(),
-		controllerContext.EventRecorder,
-	)
+	var csrController factory.Controller
+	if features.DefaultHubMutableFeatureGate.Enabled(ocmfeature.V1beta1CSRAPICompatibility) {
```

Comment:
I think we need more judgement at here, like:  
https://github.com/open-cluster-management-io/registration/blob/main/pkg/clientcert/cert_controller.go#L140